src/permissionsmodel.cpp

```diff
@@ -101,22 +113,45 @@ QVariant PermissionsModel::data(const QModelIndex &index, int role) const
     }
 }
 
-void PermissionsModel::loadPermissions()
+void PermissionsModel::clearPermissions()
 {
     if (!m_permissions.isEmpty()) {
         beginRemoveRows(QModelIndex(), 0, m_permissions.length() - 1);
         m_permissions.clear();
         endRemoveRows();
+        emit countChanged();
     }
+}
 
-    MDesktopEntry entry(m_desktopFile);
-    if (entry.isValid()) {
-        auto permissions = MPermission::fromDesktopEntry(entry);
+
+void PermissionsModel::loadPermissions()
+{
+    clearPermissions();
+    const QString appName(QFileInfo(m_desktopFile).baseName());
+    QDBusConnection systemBus(QDBusConnection::systemBus());
+    QDBusMessage request(QDBusMessage::createMethodCall(SailjailService, SailjailPath, SailjailInterface, SailjailGetAppInfo));
+    request << appName;
+    QDBusPendingReply<QVariantMap> call(systemBus.asyncCall(request));
+    QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(call, this);
+    connect(watcher, &QDBusPendingCallWatcher::finished, this, [this](QDBusPendingCallWatcher *watcher) {
+        QList<MPermission> permissions;
+        QDBusPendingReply<QVariantMap> reply = *watcher;
+        if (reply.isError()) {
+            qWarning() << SailjailGetAppInfo << "failed:" << reply.error().name() << "-" << reply.error().message();
+        } else {
+            QVariantMap appInfoMap(reply.value());
+            QStringList permissionList(appInfoMap[SailjailInfoKeyPermissions].toStringList());
+            for (const QString &permissionName : permissionList)
+                permissions.append(MPermission(PermissionFileDirectory + permissionName + PermissionSuffix));
```

Comment:
It looks like to me that this could be also `PermissionTemplate.arg(permissionName)` where `PermissionTemplate` is `"/etc/sailjail/permissions/%1.permission"` instead of concatenating a bunch `QString`s.

Reply:
done